tests/functional/UseCase/Workspace/CreateDropWorkspaceTest.php

```diff
@@ -0,0 +1,427 @@
+        $insertGroups = [
+            [
+                'columns' => '`id`',
+                'rows' => [
+                    '4',
+                    '5',
+                    '6',
+                ],
+            ],
+        ];
+
+        try {
+            $this->fillTableWithData($credentials, $bucketDatasetName, $tableName, $insertGroups);
+            $this->fail('Insert to read only table should failed');
+        } catch (ServiceException $e) {
+            $this->assertSame(403, $e->getCode());
+            $this->assertStringContainsString('Access Denied: ', $e->getMessage());
+        }
+
+        $result = $wsBqClient->runQuery($wsBqClient->query(sprintf(
+            'SELECT * FROM %s.%s;',
+            BigqueryQuote::quoteSingleIdentifier($bucketName),
+            BigqueryQuote::quoteSingleIdentifier($tableName)
+        )));
+
+        $this->assertCount(3, $result);
+        // try to create table
+        $wsBqClient->runQuery($wsBqClient->query(sprintf(
+            'CREATE TABLE %s.`testTable` (`id` INTEGER);',
+            BigqueryQuote::quoteSingleIdentifier($response->getWorkspaceObjectName())
+        )));
+
+        // try to create view
+        $wsBqClient->runQuery($wsBqClient->query(sprintf(
+            'CREATE VIEW %s.`testView` AS '
+            . 'SELECT `id` FROM %s.`testTable`;',
+            BigqueryQuote::quoteSingleIdentifier($response->getWorkspaceObjectName()),
+            BigqueryQuote::quoteSingleIdentifier($response->getWorkspaceObjectName())
+        )));
+
+        // try to drop view
+        $wsBqClient->runQuery($wsBqClient->query(sprintf(
+            'DROP VIEW %s.`testView`;',
+            BigqueryQuote::quoteSingleIdentifier($response->getWorkspaceObjectName())
+        )));
+
+        // try to drop table
+        $wsBqClient->runQuery($wsBqClient->query(sprintf(
+            'DROP TABLE %s.`testTable`;',
+            BigqueryQuote::quoteSingleIdentifier($response->getWorkspaceObjectName())
+        )));
+
+        // DROP
+        $handler = new DropWorkspaceHandler($this->clientManager);
+        $command = (new DropWorkspaceCommand())
+            ->setWorkspaceUserName($response->getWorkspaceUserName())
+            ->setWorkspaceRoleName($response->getWorkspaceRoleName())
+            ->setWorkspaceObjectName($response->getWorkspaceObjectName());
+
+        $dropResponse = $handler(
+            $this->projectCredentials,
+            $command,
+            []
+        );
+        $this->assertNull($dropResponse);
+
+        try {
+            $retryPolicy = new CallableRetryPolicy(function (Throwable $e) {
+                if ($e->getMessage() === 'Service account should be deleted.') {
+                    return true;
+                }
+                return false;
+            });
+            $proxy = new RetryProxy($retryPolicy, new ExponentialBackOffPolicy());
+            $proxy->call(function () use ($serviceAccountsService, $projectId, $wsServiceAccEmail): void {
```

Comment:
dal som sem retry pretoze google obcas trva pokial sa zpropaguje ze ten service acc sa zmazal takze to obcas failo